builds/misc/packages-release.yaml

```diff
@@ -256,11 +256,25 @@ stages:
             PathtoPublish: '$(System.ArtifactsDirectory)/$(artifactName)'
             ArtifactName: '$(artifactName)'
           condition: succeededOrFailed()
-        - task: Docker@2
-          displayName: Docker login msinternal
+
+        - task: AzureKeyVault@1
+          displayName: 'Get Secret'
           inputs:
-            command: login
-            containerRegistry: iotedge-msinternal-mcr
+            azureSubscription: $(az.subscription)
+            KeyVaultName: $(kv.name.pkgRelease)
+            SecretsFilter: >-
+              msint-spn-cert-pem
+        - pwsh: |
+            $env:MSINT_SPN_CERT_PEM | Out-File -Encoding Utf8 "$(System.ArtifactsDirectory)/msint-spn-cert.pem"
+          displayName: Install msint cert
+          env:
+            MSINT_SPN_CERT_PEM: $(msint-spn-cert-pem)
+        - bash: |
+            sudo chmod u+x "$(System.ArtifactsDirectory)/msint-spn-cert.pem"
+            az login --service-principal --use-cert-sn-issuer --username "$(servicePrincipal.clientId.msint)" --tenant "$(servicePrincipal.tenantId.msint)" -p "$(System.ArtifactsDirectory)/msint-spn-cert.pem"
```

Comment:
The [Azure CLI task](https://learn.microsoft.com/en-us/azure/devops/pipelines/tasks/deploy/azure-cli?view=azure-devops) takes a service connection. So rather than
- (AzureKeyVault@1 + service connection) to get a cert
- another task to do Azure CLI commands

...can we just use (AzureCLI@2 + service connection) directly to do the Azure CLI commands?

Reply:
No. that's the whole point of this task. The password-based i.e. ServiceConnection won't do it anymore. It is being deprecated.

Reply:
The task you're replacing here is the Docker task. So I guess you're saying they are deprecating the ability to login using service connection of type "Docker Registry with basic authentication". But AzureCLI@2 uses an [Azure Resource Manager service connection](https://learn.microsoft.com/en-us/azure/devops/pipelines/library/connect-to-azure?view=azure-devops#create-an-azure-resource-manager-service-connection-with-an-existing-service-principal), which will accept a PEM cert. Do you know if that will work? If so, it would remove KeyVault from the equation.

Reply:
It looks like it could possibly work. Though, from the MCR team's doc. They didn't provide more info on the details of the Service Principal. Only clientId & tenant. I can reach out to them for more subscription details and try to use that, but we may run into permission issues since we don't really have access to their subscription.

Reply:
Since this works, let's go with it. We can always change it if we need to.